core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java

```diff
@@ -546,7 +546,9 @@ public CreatePathResult createPath(LockedInodePath inodePath, CreatePathOptions<
     if (options.isPersisted()) {
       // Synchronously persist directories. These inodes are already READ locked.
       for (Inode inode : traversalResult.getNonPersisted()) {
-        syncPersistDirectory((InodeDirectory) inode, journalContext);
+        if (inode instanceof InodeDirectory) {
+          syncPersistDirectory((InodeDirectory) inode, journalContext);
+        }
```

Comment:
This should be an assertion? If `inode` is not an `InodeDirectory` something is wrong